Dockerfile

```diff
@@ -176,8 +176,22 @@ COPY vendor /go/src/github.com/docker/docker/vendor
 RUN go install -v github.com/cpuguy83/go-md2man \
 	github.com/BurntSushi/toml/cmd/tomlv
 
+# Build the tool for embedding resources in Windows binaries
+ENV RSRC_COMMIT e48dbf1b7fc464a9e85fcec450dddf80816b76e0
+RUN set -x \
+    && git clone https://github.com/akavel/rsrc.git /go/src/github.com/akavel/rsrc \
+    && (cd /go/src/github.com/akavel/rsrc \
+          && git checkout -q $RSRC_COMMIT \
+          && go build -v -o /go/bin/rsrc)
+
 # Wrap all commands in the "docker-in-docker" script to allow nested containers
 ENTRYPOINT ["hack/dind"]
 
 # Upload docker source
 COPY . /go/src/github.com/docker/docker
+
+# Compile the Windows resources into the sources
+RUN rsrc \
+       -manifest /go/src/github.com/docker/docker/hack/make/resources/docker.exe.manifest \
+       -ico /go/src/github.com/docker/docker/hack/make/resources/docker.ico \
+       -o   /go/src/github.com/docker/docker/pkg/winresources/rsrc.syso
```

Comment:
I used pkg\winresources as I still need the resources.go file that's there, which explains the process and provides the 'package' framework for the go package. It seemed logical that that's the right place to put it during the build process. Am open to suggestions as to where you believe an alternate place should be. Bear in mind, the binary file isn't actually ever checked in, just used during the build process.  

Note also, my end goal is to still on Windows to use Dockerfile to build on Windows in containers. Obviously that's a long way off and we need far enough developed Windows container support. But I believe it's achievable.  (FWIW, the way I currently build on Windows is a simple script which 'git clones' etc. the appropriate dependencies, effectively does what's in the Dockerfile above, go build -tags ...., deletes the dependencies and interim binary for the resources). Native building (if we go there) on Jenkins Windows will hopefully be short-lived. Until we get to that point and have made a decision, I'm not sure updating PACKAGERS.md quite yet makes sense as cross-compiling from Linux is working with this Dockerfile update currently. (That said, I still compile natively - so much easier in my workflow for daily dev work)